native/src/main/java/org/ballerinalang/redis/connection/RedisConnectionCommandExecutor.java

```diff
@@ -33,29 +38,37 @@
 public class RedisConnectionCommandExecutor {
 
     private final RedisConnectionManager<?, ?> connManager;
+    private static final String CLUSTER_INFO_SEPARATOR = "\\r\\n";
 
     public RedisConnectionCommandExecutor(RedisConnectionManager<?, ?> connManager) {
         this.connManager = connManager;
     }
 
     public String auth(String password) throws RedisConnectorException {
+        RedisClusterCommands<?, String> clusterCommands = null;
         RedisCommands<?, String> redisCommands = null;
         try {
-            redisCommands = (RedisCommands<?, String>) connManager.getConnectionCommandConnection();
-            return redisCommands.auth(password);
+            if (connManager.isClusterConnection()) {
+                clusterCommands = (RedisAdvancedClusterCommands<?, String>) connManager.getRedisClusterCommands();
+                return clusterCommands.auth(password);
+            } else {
+                redisCommands = (RedisCommands<?, String>) connManager.getRedisCommands();
+                return redisCommands.auth(password);
+            }
         } catch (IllegalArgumentException e) {
             throw new RuntimeException("Password " + MUST_NOT_BE_NULL, e);
         } catch (RedisException e) {
             throw new RedisConnectorException(REDIS_SERVER_ERROR + e.getMessage(), e);
         } finally {
+            connManager.releaseResources(clusterCommands);
             connManager.releaseResources(redisCommands);
         }
     }
 
-    public String echo(String message) throws RedisConnectorException {
-        RedisCommands<?, String> redisCommands = null;
+    public <K> String echo(String message) throws RedisConnectorException {
+        BaseRedisCommands<K, String> redisCommands = null;
         try {
-            redisCommands = (RedisCommands<?, String>) connManager.getConnectionCommandConnection();
+            redisCommands = (BaseRedisCommands<K, String>) connManager.getConnectionCommandConnection();
```

Comment:
If `null` initialized `redisCommand` is only used under try we can initialize under try itself, isn't it?

```suggestion
            BaseRedisCommands<K, String> redisCommands = (BaseRedisCommands<K, String>) connManager.getConnectionCommandConnection();
```

Reply:
@AzeemMuzammil we cannot do this as `redisCommands` is used inside the `finally` block as well.